hadoop-yarn-project/hadoop-yarn/hadoop-yarn-server/hadoop-yarn-server-common/src/main/java/org/apache/hadoop/yarn/server/federation/utils/FederationRegistryClient.java

```diff
@@ -203,21 +203,27 @@ public synchronized boolean writeAMRMTokenForUAM(ApplicationId appId,
    * Remove an application from registry.
    *
    * @param appId application id
+   * @param ignoreMemoryState whether to ignore the memory data in terms of
+   *          known application
    */
-  public synchronized void removeAppFromRegistry(ApplicationId appId) {
+  public synchronized void removeAppFromRegistry(ApplicationId appId,
+      boolean ignoreMemoryState) {
```

Comment:
Can we have a default method with this as false (or true, whatever is the common case), to prevent changing the signature?

Reply:
I will improve this part of the code.